src/metabase/routes.clj

```diff
@@ -40,9 +41,17 @@
   ;; ^/$ -> index.html
   (GET "/" [] index)
   (GET "/favicon.ico" [] (resp/resource-response "frontend_client/favicon.ico"))
-  ;; ^/api/ -> API routes
-  (context "/api" [] api/routes)
-  ; ^/app/ -> static files under frontend_client/app
+  ;; ^/api/health -> Health Check Endpoint
+  (GET "/api/health" [] (if (init-status/complete?)
+                          {:status 200, :body {:status "ok"}}
+                          {:status 503, :body {:status "initializing", :progress (init-status/progress)}}))
+  ;; ^/api/ -> All other API routes
+  (context "/api" [] (fn [& args]
+                       ;; if Metabase is not finished initializing, return a generic error message rather than something potentially confusing like "DB is not set up"
+                       (if-not (init-status/complete?)
+                         {:status 400, :body "Metabase is still initializing. Please sit tight..."}
```

Comment:
This should probably return 503 like `/api/health`. A 4xx client error seems inappropriate.